src/main/java/com/google/devtools/build/lib/bazel/rules/python/BazelPyRuleClasses.java

```diff
@@ -176,6 +177,11 @@ public RuleClass build(RuleClass.Builder builder, final RuleDefinitionEnvironmen
               .allowedFileTypes(PYTHON_SOURCE)
               .direct_compile_time_input()
               .allowedFileTypes(BazelPyRuleClasses.PYTHON_SOURCE))
+          /* <!-- #BLAZE_RULE($base_py_binary).ATTRIBUTE(legacy_create_init) -->
+          Controls whether to create __init__.py files in paths where they don't exists
+          or not, default is to true for backward compatibility and to support legacy behavior.
+          <!-- #END_BLAZE_RULE.ATTRIBUTE --> */
+          .add(attr("legacy_create_init", BOOLEAN).value(true))
```

Comment:
Please advise on the name of the attribute, I went with the `legacy_` convention since I saw it sue for some attributes but I am not sure if it's the right use case.

Reply:
I was wondering if this should be an enum or integer, in case we want more than two behaviors.  But I can't think of any third thing we might do.

Can we call it legacy if we don't have a specific plan to switch everything to the new behavior?  I'm ok with it, but if you wanted a different name, maybe `implict_create_init_py`?

You could expand the comment a bit to explain the legacy behavior:

```
Whether to implicitly create empty __init__.py files in the runfiles tree.
These are created in every directory containing Python source code or
shared libraries, and every parent directory of those directories.  
Default is true for backward compatibility.  If false, the user is responsible
for creating __init__.py files (empty or not) and adding them to `srcs` or `deps`
of Python targets as required.
```

Reply:
I like your description will adapt, concerning the attribute name IMHO we should drop the behavior in the future and that's how I understood it here https://github.com/bazelbuild/rules_python/issues/55#issuecomment-354638528.